tools/manifest/vcs.py

```diff
@@ -111,15 +116,27 @@ def __init__(self,
 
     def __iter__(self) -> Iterator[Tuple[Text, Optional[Text], bool]]:
         mtime_cache = self.mtime_cache
-        for dirpath, dirnames, filenames in self.path_filter(
-                walk(self.tests_root.encode("utf8"))):
-            for filename, path_stat in filenames:
-                path = os.path.join(dirpath, filename).decode("utf8")
+        for subdir in self.subdirs_to_update:
+            path = os.path.join(self.tests_root, subdir)
+            if os.path.isfile(path):
+                path_stat = os.stat(path)
+                path = subdir
                 if mtime_cache is None or mtime_cache.updated(path, path_stat):
                     file_hash = self.hash_cache.get(path, None)
                     yield path, file_hash, True
                 else:
                     yield path, None, False
+            elif os.path.isdir(path):
+                for dirpath, dirnames, filenames in self.path_filter(
+                        walk(path.encode("utf8"))):
+                    for filename, path_stat in filenames:
+                        path = os.path.join(subdir,
+                                            os.path.join(dirpath, filename).decode("utf8"))
+                        if mtime_cache is None or mtime_cache.updated(path, path_stat):
```

Comment:
nit: Put the shared file case in a internal helper (e.g., `yield self._make_file_info(path, path_stat)`)?